registry/remote/registry_test.go

```diff
@@ -194,3 +194,45 @@ func TestRegistry_Repository(t *testing.T) {
 		t.Errorf("Registry.Repository() = %v, want %v", got, want)
 	}
 }
+
+func testRegistry() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		result := struct {
+			Repositories []string `json:"repositories"`
+		}{
+			Repositories: []string{"public/repo1", "public/repo2", "internal/repo3"},
+		}
+		json.NewEncoder(w).Encode(result)
+	}))
+}
+
+// This is a example for listing respositories in the registry:
+func ExampleRegistry_Repositories() {
+	// Mocking local registry
+	ts := testRegistry()
+	defer ts.Close()
+	uri, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err)
+	}
+
+	// Example: List repositories in a registry
+	reg, err := NewRegistry(uri.Host) // Create a registry via the remote host
+	if err != nil {
+		panic(err) // Handle error
+	}
+	reg.RepositoryOptions.PlainHTTP = true // Use HTTP
```

Comment:
This can be simplified as `reg.PlainHTTP = true`.

Reply:
thanks for pointing out, but why?